applications/mlflow/DEVELOPMENT.md

```diff
@@ -2,42 +2,106 @@
+
+Optional (for direct installation on your machine):
+- go (1.20+)
+- Helm
+- kubectl
+
 ## Development Workflow
 
-The MLflow application includes a Taskfile.yml that provides tasks for developing, testing, and publishing the application.
+We use a [Taskfile](https://taskfile.dev/) to manage development tasks. The recommended approach is to use our containerized development environment, which includes all necessary dependencies.
+
+### Containerized Development Environment
+
+Our Docker-based development environment provides:
+* Consistent environment across team members
+* All required dependencies pre-installed
+* No need to install Go, Python, Helm, or other tools locally
+* Works on any operating system with Docker support
+
+To get started:
+
+```bash
+# Build the development image
+task dev:build-image
+
+# Start a development container
+# This creates and runs the container in the background
+task dev:start
+
+# Enter a shell in an already running development container
+task dev:shell
+```
+
+The tasks above will:
+1. Build the development Docker image if needed
+2. Start a container with the proper mounts and environment
+3. Provide you with a shell inside the container
+4. Allow you to run all task commands directly
+
+### Common Development Tasks
+
+Once you're inside the development container (or on your local machine with all prerequisites installed), you can run these common tasks:
+
+```bash
+# Update Helm dependencies
+task helm:update-deps
+
+# Lint the Helm chart
+task helm:lint
+
+# Template the chart (no values overrides)
+task helm:template
 
-### Prerequisites
+# Install the chart to your Kubernetes cluster
+task helm:install-local
 
-- [Task](https://taskfile.dev/#/installation) command line tool
-- Kubernetes cluster configured in your current context
-- kubectl, helm, and python3 installed
+# Run application tests
+task test:app
+```
 
-### Local Development
+### Development Workflow Steps
 
-Follow this workflow for development:
+1. Set up the development environment:
+   ```bash
+   task dev:build-image
+   ```
+
+2. Start the development container:
+   ```bash
+   task dev:start
+   ```
 
-1. Add required Helm repositories and update dependencies:
+3. Enter the development container:
    ```bash
-   task update:deps:helm
+   task dev:shell
    ```
 
-2. Lint charts to check for issues:
+4. Update Helm dependencies:
    ```bash
-   task lint
+   task helm:update-deps
    ```
 
-3. Template charts to verify the rendered manifests:
+5. Lint and template charts to check for issues:
    ```bash
-   task template
+   task helm:lint
+   task helm:template
    ```
 
-4. Install charts for development:
+6. Install charts for development:
    ```bash
    # Installs with Replicated SDK disabled
-   task install:helm:local
+   task helm:install-local
    
    # Optionally specify a custom values file
-   MLFLOW_VALUES=./my-values.yaml task install:helm:local
+   MLFLOW_VALUES=./my-values.yaml task helm:install-local
```

Comment:
Is this setup so that you can provide different values to different charts? As a generic workflow I'd like to see inputs and configs like this not include the application name. To make this a process people can use in their own applications we want to minimize the number of variables where the variable name is specific to an application. Ideally the workflow has all generic variable names and works for any application suite, and the contents of the variables is the only custom parts.